test/classes/nilability/nil-non-subtype.chpl

```diff
@@ -20,7 +20,7 @@ proc test(type A, type B) {
 
 proc main() {
   test(C, C?);
-  test(G, G?);
+  test(G(?), G?(?));
```

Comment:
Do we permit `G?(?)` and `G(?)?`?  I would've imagined only the latter...

Reply:
Actually I updated this test in this way accidentally, but I believe we permit both. The reason is that the code to determine if `(?)` is in use is pretty hacky and based on flags set in the normalizer (rather than, say, having a different type for `G` and `G(?)` that flows through resolution).